kong/db/migrations/core/007_140_to_200.lua

```diff
@@ -23,7 +95,67 @@ return {
   },
 
   cassandra = {
-    up = [[]],
+    up = function(connector)
+      local cql = "ALTER TABLE routes ADD path_handling text";
+      local res, err = connector:query(cql);
+      if not res then
+        if connector:is_ignorable_during_migrations(err) then
+          ngx.log(ngx.WARN, fmt(
+            "ignored error while running '007_140_to_200' migration: %s (%s)",
+            err, cql
+          ))
+        else
+          error(err)
+        end
+      end
+
+      local rows = assert(connector:query([[
+        SELECT last_executed FROM schema_meta
+          WHERE key = ? AND subsystem = ?
+      ]], {
+        cassandra.text("007_140_to_200"),
+        cassandra.text("path_handling")
+      }))
+      local preset_path_handling = rows and rows[1] and rows[1].last_executed
+
+      if not preset_path_handling then
+        local rows = assert(connector:query([[
+          SELECT last_executed FROM schema_meta
+            WHERE key = ? AND subsystem = ?
+        ]], {
+          cassandra.text("schema_meta"),
+          cassandra.text("core")
+        }))
+        local migrating_from = rows and rows[1] and rows[1].last_executed
+
+        if migrating_from then
+          preset_path_handling = PATH_HANDLING_WHEN_MIGRATING_FROM[migrating_from]
+        end
+      end
+
+      if preset_path_handling then
+        assert(connector:query([[
+          INSERT INTO schema_meta (key, subsystem, last_executed)
+            VALUES(?, ?, ?)
+        ]], {
+          cassandra.text("007_140_to_200"),
+          cassandra.text("path_handling"),
+          cassandra.text(preset_path_handling)
+        }))
+
+        local rows = assert(connector:query([[
+          SELECT id FROM routes;
+        ]]))
```

Comment:
This does *not* give you all routes in Cassandra, you must use the `iterate` method.